### PR TITLE
Update CI_YAML.MD regarding taget updates

### PR DESCRIPTION
**Pull request by keyonghan**
Following https://github.com/flutter/flutter/pull/120912#discussion_r1128793254 to make doc clear.

### CODE REVIEW

File: CI_YAML.md [hunk 1]
```diff
@@ -165,11 +167,14 @@ tags: >
               {"dependency": "open_jdk", "version": "11"}
           ]
         timeout: 60
-    ```
-    - Send PR, wait for the checks to go green (the change takes effect on presubmit)
+      ```
+    - Send PR, wait for the checks to go green (**the change takes effect on presubmit**)
 3. If the check is red, add patches to get it green
 4. Once the PR has landed, infrastructure may take 1 or 2 commits to apply the latest properties
```
Comment (stuartmorgan) [inline CI_YAML.md:173]: I'm confused; it takes effect immediately for presubmit, but not for postsubmit? So property changes still can't be landed in the same PR as changes that require them (like removing bring up)?

Or is this not the case for `properties`?

Comment (stuartmorgan) [reply CI_YAML.md:173]: So item 4 here should be removed, right?

Comment (keyonghan) [reply CI_YAML.md:174]: We still need it. The `properties` will be seen by the current PR/commit immediately, but there are cases where we need to wait, such incoming PRs/commits, which haven't rebased on top of that changing PR, and local LED runs.

Expanded a bit. Please let me know if this makes sense to you.

**Review by stuartmorgan — APPROVED** [review]
LGTM

I don't understand why something that hasn't rebased would need to "wait 1 or 2 commits" rather than having to rebase, but I'll assume that's me not understanding the infrastructure.

**Comment by keyonghan** [comment]
> LGTM
> 
> I don't understand why something that hasn't rebased would need to "wait 1 or 2 commits" rather than having to rebase, but I'll assume that's me not understanding the infrastructure.

Suppose we have a property change PR (pr1) and an incoming PR (pr2):
Note that cocoon schedules builds based on the current commit `sha`.
If a PR (pr2) hasn't rebased, cocoon will use pr2's target properties (without changes from pr1) on top of current infrastructure configs (pr1's changes either propagated or not) to schedule. If the pr1's changes haven't been propagated, they will not be reflected in pr2's scheduled builds. Yeah, this is some backend infra logics:-)

**Comment by stuartmorgan** [comment]
> on top of current infrastructure configs (pr1's changes either propagated or not)

Wait, so does the infra behavior distinguish between "the state of .ci.yaml in the PR" and "the *changes* to .ci.yaml in the PR". I.e., looking at deltas in the PR specifically?

**Comment by keyonghan** [comment]
> > on top of current infrastructure configs (pr1's changes either propagated or not)
> 
> Wait, so does the infra behavior distinguish between "the state of .ci.yaml in the PR" and "the _changes_ to .ci.yaml in the PR". I.e., looking at deltas in the PR specifically?

That's correct. If no changes in `.ci.yaml`, cocoon will trigger builds with whatever current infra configs are. If any changes, cocoon will fetch them and override corresponding values in existing infra configs. That's why the changes (properties) in the PR could be reflected in the current PR right away.

File: CI_YAML.md (hunk 1 above)
Comment (CaseyHillers) [reply CI_YAML.md:174]: Another case is caches. While dependencies can be upgraded, there may be some slow downs on the initial roll out due to cache invalidation

Comment (keyonghan) [reply CI_YAML.md:174]: Added one more item to cover.

**Pull request merged**